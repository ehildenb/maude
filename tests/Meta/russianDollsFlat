#!/bin/sh

MAUDE_LIB=$srcdir/../../src/Main
export MAUDE_LIB

../../src/Main/maude \
  < $srcdir/russianDollsFlat.maude -no-banner -no-advise \
  > russianDollsFlat.out 2>&1

diff $srcdir/russianDollsFlat.expected russianDollsFlat.out > /dev/null 2>&1
