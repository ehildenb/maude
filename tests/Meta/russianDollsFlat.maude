set show timing off .
set show advisories off .

***
***	Run nested meta-interpreters with flat meta-modules.
***

load metaInterpreter

mod RUSSIAN-DOLLS is
  pr META-INTERPRETER .

  op me : -> Oid .
  op User : -> Cid .
  op level:_ : Nat -> Attribute .

  vars X Y Z : Oid .
  var AS : AttributeSet .
  var N : Nat .

  op problem : Nat -> Term .
  eq problem(s N) =
  '__['<>.Portal,'createInterpreter['interpreterManager.Oid,'me.Oid,'none.InterpreterOptionSet],
    '<_:_|_>['me.Oid, 'User.Cid, 'level:_[upTerm(N)]]] .
  eq problem(0) = '_+_['s_^2['0.Zero], 's_^2['0.Zero]] .

  rl < X : User | AS > createdInterpreter(X, Y, Z) =>
     < X : User | AS > insertModule(Z, X, upModule('RUSSIAN-DOLLS, true)) .

  rl < X : User | level: N, AS > insertedModule(X, Y) =>
     < X : User | AS > erewriteTerm(Y, X, unbounded, 1, 'RUSSIAN-DOLLS, problem(N)) .
endm

erew in RUSSIAN-DOLLS : <> 
< me : User | level: 0 > 
createInterpreter(interpreterManager, me, none) .

erew in RUSSIAN-DOLLS : <> 
< me : User | level: 1 > 
createInterpreter(interpreterManager, me, none) .

erew in RUSSIAN-DOLLS : <> 
< me : User | level: 2 > 
createInterpreter(interpreterManager, me, none) .

erew in RUSSIAN-DOLLS : <> 
< me : User | level: 3 > 
createInterpreter(interpreterManager, me, none) .

erew in RUSSIAN-DOLLS : <> 
< me : User | level: 4 > 
createInterpreter(interpreterManager, me, none) .
