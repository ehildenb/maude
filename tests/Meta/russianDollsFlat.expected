==========================================
erewrite in RUSSIAN-DOLLS : <> < me : User | level: 0 > createInterpreter(
    interpreterManager, me, none) .
rewrites: 6
result Configuration: <> < me : User | none > erewroteTerm(me, interpreter(0),
    1, 's_^4['0.Zero], 'NzNat)
==========================================
erewrite in RUSSIAN-DOLLS : <> < me : User | level: 1 > createInterpreter(
    interpreterManager, me, none) .
rewrites: 11
result Configuration: <> < me : User | none > erewroteTerm(me, interpreter(0),
    5, '__['<>.Portal,'<_:_|_>['me.Oid,'User.Cid,'none.AttributeSet],
    'erewroteTerm['me.Oid,'interpreter['0.Zero],'s_['0.Zero],'_`[_`][
    ''s_^4.Sort,''0.Zero.Constant],''NzNat.Sort]], 'Configuration)
==========================================
erewrite in RUSSIAN-DOLLS : <> < me : User | level: 2 > createInterpreter(
    interpreterManager, me, none) .
rewrites: 16
result Configuration: <> < me : User | none > erewroteTerm(me, interpreter(0),
    10, '__['<>.Portal,'<_:_|_>['me.Oid,'User.Cid,'none.AttributeSet],
    'erewroteTerm['me.Oid,'interpreter['0.Zero],'s_^5['0.Zero],'_`[_`][
    ''__.Sort,'_`,_[''<>.Portal.Constant,'_`[_`][''<_:_|_>.Variable,'_`,_[
    ''me.Oid.Constant,''User.Cid.Constant,''none.AttributeSet.Constant]],
    '_`[_`][''erewroteTerm.Sort,'_`,_[''me.Oid.Constant,'_`[_`][
    ''interpreter.Sort,''0.Zero.Constant],'_`[_`][''s_.Sort,''0.Zero.Constant],
    '_`[_`][''_`[_`].Qid,'_`,_['''s_^4.Sort.Constant,
    '''0.Zero.Constant.Constant]],'''NzNat.Sort.Constant]]]],
    ''Configuration.Sort]], 'Configuration)
==========================================
erewrite in RUSSIAN-DOLLS : <> < me : User | level: 3 > createInterpreter(
    interpreterManager, me, none) .
rewrites: 21
result Configuration: <> < me : User | none > erewroteTerm(me, interpreter(0),
    15, '__['<>.Portal,'<_:_|_>['me.Oid,'User.Cid,'none.AttributeSet],
    'erewroteTerm['me.Oid,'interpreter['0.Zero],'s_^10['0.Zero],'_`[_`][
    ''__.Sort,'_`,_[''<>.Portal.Constant,'_`[_`][''<_:_|_>.Variable,'_`,_[
    ''me.Oid.Constant,''User.Cid.Constant,''none.AttributeSet.Constant]],
    '_`[_`][''erewroteTerm.Sort,'_`,_[''me.Oid.Constant,'_`[_`][
    ''interpreter.Sort,''0.Zero.Constant],'_`[_`][''s_^5.Sort,
    ''0.Zero.Constant],'_`[_`][''_`[_`].Qid,'_`,_['''__.Sort.Constant,'_`[_`][
    ''_`,_.Qid,'_`,_['''<>.Portal.Constant.Constant,'_`[_`][''_`[_`].Qid,'_`,_[
    '''<_:_|_>.Variable.Constant,'_`[_`][''_`,_.Qid,'_`,_[
    '''me.Oid.Constant.Constant,'''User.Cid.Constant.Constant,
    '''none.AttributeSet.Constant.Constant]]]],'_`[_`][''_`[_`].Qid,'_`,_[
    '''erewroteTerm.Sort.Constant,'_`[_`][''_`,_.Qid,'_`,_[
    '''me.Oid.Constant.Constant,'_`[_`][''_`[_`].Qid,'_`,_[
    '''interpreter.Sort.Constant,'''0.Zero.Constant.Constant]],'_`[_`][
    ''_`[_`].Qid,'_`,_['''s_.Sort.Constant,'''0.Zero.Constant.Constant]],
    '_`[_`][''_`[_`].Qid,'_`,_['''_`[_`].Qid.Constant,'_`[_`][''_`,_.Qid,'_`,_[
    ''''s_^4.Sort.Constant.Constant,''''0.Zero.Constant.Constant.Constant]]]],
    ''''NzNat.Sort.Constant.Constant]]]]]]]],
    '''Configuration.Sort.Constant]]]],''Configuration.Sort]], 'Configuration)
==========================================
erewrite in RUSSIAN-DOLLS : <> < me : User | level: 4 > createInterpreter(
    interpreterManager, me, none) .
rewrites: 26
result Configuration: <> < me : User | none > erewroteTerm(me, interpreter(0),
    20, '__['<>.Portal,'<_:_|_>['me.Oid,'User.Cid,'none.AttributeSet],
    'erewroteTerm['me.Oid,'interpreter['0.Zero],'s_^15['0.Zero],'_`[_`][
    ''__.Sort,'_`,_[''<>.Portal.Constant,'_`[_`][''<_:_|_>.Variable,'_`,_[
    ''me.Oid.Constant,''User.Cid.Constant,''none.AttributeSet.Constant]],
    '_`[_`][''erewroteTerm.Sort,'_`,_[''me.Oid.Constant,'_`[_`][
    ''interpreter.Sort,''0.Zero.Constant],'_`[_`][''s_^10.Sort,
    ''0.Zero.Constant],'_`[_`][''_`[_`].Qid,'_`,_['''__.Sort.Constant,'_`[_`][
    ''_`,_.Qid,'_`,_['''<>.Portal.Constant.Constant,'_`[_`][''_`[_`].Qid,'_`,_[
    '''<_:_|_>.Variable.Constant,'_`[_`][''_`,_.Qid,'_`,_[
    '''me.Oid.Constant.Constant,'''User.Cid.Constant.Constant,
    '''none.AttributeSet.Constant.Constant]]]],'_`[_`][''_`[_`].Qid,'_`,_[
    '''erewroteTerm.Sort.Constant,'_`[_`][''_`,_.Qid,'_`,_[
    '''me.Oid.Constant.Constant,'_`[_`][''_`[_`].Qid,'_`,_[
    '''interpreter.Sort.Constant,'''0.Zero.Constant.Constant]],'_`[_`][
    ''_`[_`].Qid,'_`,_['''s_^5.Sort.Constant,'''0.Zero.Constant.Constant]],
    '_`[_`][''_`[_`].Qid,'_`,_['''_`[_`].Qid.Constant,'_`[_`][''_`,_.Qid,'_`,_[
    ''''__.Sort.Constant.Constant,'_`[_`][''_`[_`].Qid,'_`,_[
    '''_`,_.Qid.Constant,'_`[_`][''_`,_.Qid,'_`,_[
    ''''<>.Portal.Constant.Constant.Constant,'_`[_`][''_`[_`].Qid,'_`,_[
    '''_`[_`].Qid.Constant,'_`[_`][''_`,_.Qid,'_`,_[
    ''''<_:_|_>.Variable.Constant.Constant,'_`[_`][''_`[_`].Qid,'_`,_[
    '''_`,_.Qid.Constant,'_`[_`][''_`,_.Qid,'_`,_[
    ''''me.Oid.Constant.Constant.Constant,
    ''''User.Cid.Constant.Constant.Constant,
    ''''none.AttributeSet.Constant.Constant.Constant]]]]]]]],'_`[_`][
    ''_`[_`].Qid,'_`,_['''_`[_`].Qid.Constant,'_`[_`][''_`,_.Qid,'_`,_[
    ''''erewroteTerm.Sort.Constant.Constant,'_`[_`][''_`[_`].Qid,'_`,_[
    '''_`,_.Qid.Constant,'_`[_`][''_`,_.Qid,'_`,_[
    ''''me.Oid.Constant.Constant.Constant,'_`[_`][''_`[_`].Qid,'_`,_[
    '''_`[_`].Qid.Constant,'_`[_`][''_`,_.Qid,'_`,_[
    ''''interpreter.Sort.Constant.Constant,
    ''''0.Zero.Constant.Constant.Constant]]]],'_`[_`][''_`[_`].Qid,'_`,_[
    '''_`[_`].Qid.Constant,'_`[_`][''_`,_.Qid,'_`,_[
    ''''s_.Sort.Constant.Constant,''''0.Zero.Constant.Constant.Constant]]]],
    '_`[_`][''_`[_`].Qid,'_`,_['''_`[_`].Qid.Constant,'_`[_`][''_`,_.Qid,'_`,_[
    ''''_`[_`].Qid.Constant.Constant,'_`[_`][''_`[_`].Qid,'_`,_[
    '''_`,_.Qid.Constant,'_`[_`][''_`,_.Qid,'_`,_[
    '''''s_^4.Sort.Constant.Constant.Constant,
    '''''0.Zero.Constant.Constant.Constant.Constant]]]]]]]],
    '''''NzNat.Sort.Constant.Constant.Constant]]]]]]]]]]]]]]]],
    ''''Configuration.Sort.Constant.Constant]]]]]]]],
    '''Configuration.Sort.Constant]]]],''Configuration.Sort]], 'Configuration)
Bye.
